test/mt_id_test.c

```diff
@@ -6328,11 +6369,12 @@ void serial_test_1(void)
 
     if ( 0 == err_cnt ) {
 
-         PASSED();
+        PASSED();
 
     } else {
 
-         H5_FAILED();
+        IncTestNumErrs();
```

Comment:
This is a little more coarse-grained than I'd like, but the testing framework will likely need a little revision to better capture the true number of errors that occurred. For now, this is just to signal if even one error occurred so that we can eventually exit with the appropriate exit status value.

Reply:
From @mattjala:

> Are there any issues with directly translating increments of `err_cnt` to `IncTestNumErrs()` besides potential concurrent access?

Reply:
The main issue I ran into is that some of the increments of `err_cnt` look something like `err_cnt += func()`, opening the possibility for incrementing by more than 1. We could add a count parameter to `IncTestNumErrs()`, but I didn't want to add to the noise with that change yet. The error count in the testing framework also is directly being incremented with `++` even though it's now an atomic variable, so we should change that to either an atomic increment or atomic add (if we add a count parameter). Then of course there may be some issues around making sure the error count value makes sense after concurrent increments.